qt/widgets/instrumentview/test/InstrumentWidget/InstrumentWidgetTest.h

```diff
@@ -262,26 +306,37 @@ class InstrumentWidgetTest : public CxxTest::TestSuite {
     return mock;
   }
 
-  InstrumentWidget construct(std::unique_ptr<DisplayMock> displayMock, QtMock *qtMock, GLMock *glMock,
-                             std::unique_ptr<ConnectMock> connectMock,
-                             std::unique_ptr<MetaObjectMock> metaObjectMock) const {
+  class MockInstrumentWidgetMaskTab final : public InstrumentWidgetMaskTab {
+  public:
+    explicit MockInstrumentWidgetMaskTab(InstrumentWidget *instrWidget) : InstrumentWidgetMaskTab(instrWidget){};
+
+    GNU_DIAG_OFF_SUGGEST_OVERRIDE
+    MOCK_METHOD1(showMessageBox, void(const QString &));
```

Comment:
Since this is a new mock can we use the new mocking decelerations that doesn't need warning suppressions

```c++
MOCK_METHOD(void, showMessageBox, (const QString &), (override))
```

This allows us to drop warning suppressions, and will catch cases where the signature changes but this mock doesn't (currently this will not produce a warning, only a runtime error)